runner/__init__.py

```diff
@@ -81,6 +91,19 @@ def pytest_runtest_logreport(self, report):
         source = Path(self.config.rootdir) / report.fspath
         state.test_code = TestOrder.function_source(test_id, source)
 
+        # Looks up tast_ids from parent when the test is a subtest.
+        if state.task_id == 0 and 'variation' in state.name:
+            parent_test_name = state.name.split(' ')[0]
+            parent_task_id = self.tests[parent_test_name].task_id
+            state.task_id = parent_task_id
+
+            # Changes status of parent to fail if any of the subtests fail.
+            if state.fail:
+                self.tests[parent_test_name].fail(
+                    message="One or more subtests for this test failed. Details can be found under each variant."
+                )
+                self.tests[parent_test_name].test_code = state.test_code
```

Comment:
Does the subtests module not already do this?

Reply:
🤣    No.  The subtests module makes it possible to use `subttest` without PyTest freaking out.  There is actually [an ongoing discussion](https://github.com/pytest-dev/pytest-subtests/issues/11) on the `pytest-subtest` repo on how to handle failures and test counting right now.   And - as the discussion outlines - this is actually behavior inherited from `unittest`.  So for our plugin, I decided to make it "cleaner".  But it is still weird to have a count mis-match when all tests pass vs when some that have sub-tests fail (_this happens with all parameterization, in Unittest/Pytest if I am remembering correctly_).

Reply:
...I'd actually love to do a refactor of our runner that treats parameterization more like a matrix -- so that there is a clean count of which tests are parameterized, then a process that "explodes" the matrix into individual cases, then makes a summary.  But I don't think we want to do that right now.

Reply:
No probably right now, but that sounds like a good idea.